scripts/jest/utils/index.ts

```diff
@@ -103,12 +114,17 @@ function toMatchFile(receivedContent, filename) {
     }
 }
 
-/**
- * @typedef {Object} TestFixtureConfig
- * @property {string} src - the fixture content
- * @property {string} filename - the fixture absolute path
- * @property {string} dirname - the fixture directory absolute path
- */
+declare global {
+    namespace jest {
+        interface Matchers<R> {
+            __type: R; // unused, but makes TypeScript happy
+            toMatchFile(receivedContent: string, filename?: string): CustomMatcherResult;
+        }
+    }
+}
```

Comment:
Use the vue-next approach for this.

Reply:
See comment above for https://github.com/salesforce/lwc/pull/2427#discussion_r674755704